src/plugins/image_embeddable/public/components/image_embeddable.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { ReactEmbeddableDynamicActionsApi } from '@kbn/embeddable-enhanced-plugin/public/plugin';
+import { PublishingSubject, useStateFromPublishingSubject } from '@kbn/presentation-publishing';
+
+import { imageClickTrigger } from '../actions';
+import { ImageEmbeddableStrings } from '../image_embeddable/image_embeddable_strings';
+import { ImageEmbeddableApi } from '../image_embeddable/types';
+import { FileImageMetadata, FilesClient, imageEmbeddableFileKind } from '../imports';
+import { coreServices, screenshotModeService, uiActionsService } from '../services/kibana_services';
+import { ImageConfig } from '../types';
+import { createValidateUrl } from '../utils/validate_url';
+import { ImageViewer } from './image_viewer';
+import { ImageViewerContext } from './image_viewer/image_viewer_context';
+
+import './image_embeddable.scss';
+
+interface ImageEmbeddableProps {
+  api: ImageEmbeddableApi & {
+    setBlockingError: (error: Error | undefined) => void;
+    setDataLoading: (loading: boolean | undefined) => void;
+    imageConfig$: PublishingSubject<ImageConfig>;
+  };
+  filesClient: FilesClient<FileImageMetadata>;
+  startDynamicActions?: ReactEmbeddableDynamicActionsApi['startDynamicActions'];
+}
+
+export const ImageEmbeddable = ({
+  api,
+  filesClient,
+  startDynamicActions,
+}: ImageEmbeddableProps) => {
+  const imageConfig = useStateFromPublishingSubject(api.imageConfig$);
+  const dynamicActionsState = useStateFromPublishingSubject(api.dynamicActionsState$);
+
+  const [hasTriggerActions, setHasTriggerActions] = useState(false);
+
+  useEffect(() => {
+    if (!startDynamicActions) return;
+    // start the dynamic actions manager on mount
+    const { stopDynamicActions } = startDynamicActions();
+    return () => {
+      // stop the dynamic actions manager on unmount
+      stopDynamicActions();
+    };
+  }, [startDynamicActions]);
+
+  useEffect(() => {
+    setHasTriggerActions((dynamicActionsState?.dynamicActions.events ?? []).length > 0);
+  }, [dynamicActionsState]);
+
+  return (
+    <ImageViewerContext.Provider
+      value={{
+        getImageDownloadHref: (fileId: string) => {
+          return filesClient.getDownloadHref({
+            id: fileId,
+            fileKind: imageEmbeddableFileKind.id,
+          });
+        },
+        validateUrl: createValidateUrl(coreServices.http.externalUrl),
+      }}
+    >
+      <ImageViewer
+        data-rendering-count={1} // TODO: Remove this as part of https://github.com/elastic/kibana/issues/179376
```

Comment:
This is normally updated whenever data is fetched - so, since the image embeddable never fetches data, a constant `1` is valid and consistent with previous behaviour.